src/zero/once.clj

```diff
@@ -26,8 +30,48 @@
 ;;
 (defonce the-cached-credentials (delay (new-user-credentials)))
 
-(defn get-local-auth-header
-  "Return a valid auth header, refresh the access token
-  under the hood if necessary."
+(defn new-credentials-from-service-account
+  "Generate scoped GoogleCredentials from a service account FILE."
+  [^String file]
+  (when file
+    (let [scopes ["email" "profile" "openid"]
+          credentials (GoogleCredentials/fromStream (FileInputStream. file))]
+      (.createScoped credentials scopes))))
+
+(defn service-account-for-env
+  "Ge the path to service account for ENVIRONMENT."
+  [environment]
+  (get-in env/stuff [environment :cromwell :wfl-service-account]))
+
+;; The re-usable environment:credentials-object map for token generation and refreshing
+;;
+(defonce the-cached-credentials-from-service-account
+         (delay
+           (let [env   (zero/throw-or-environment-keyword! (System/getenv "ENVIRONMENT"))]
+             (new-credentials-from-service-account (service-account-for-env env)))))
+
+(defn get-auth-header!
+  "Return a valid auth header. Refresh and generate the access
+   token with gcloud command if invoked from command line,
+   generate the access token from service account if invoked
+   from a live server."
   []
-  (util/bearer-token-header-for @the-cached-credentials))
+  (if (and (System/getenv "WFL_LIVE_SERVER_MODE") (System/getenv "ENVIRONMENT"))
+    (util/bearer-token-header-for @the-cached-credentials-from-service-account)
+    (util/bearer-token-header-for @the-cached-credentials)))
```

Comment:
The `and` is a little weird here.  What does `"ENVIRONMENT"` add?

Reply:
It does not add anything, I was thinking making the conditions for `@the-cached-credentials-from-service-account` explicit (both `WFL_LIVE_SERVER_MODE ` and `ENVIRONMENT ` have to be set properly) was a good thing 🤔